src/provider/provider_tracking.c

```diff
@@ -161,11 +162,35 @@ static umf_result_t trackingAlloc(void *hProvider, size_t size,
         return ret;
     }
 
-    umf_result_t ret2 = umfMemoryTrackerAdd(p->hTracker, p->pool, *ptr, size);
-    if (ret2 != UMF_RESULT_SUCCESS) {
-        LOG_ERR("failed to add allocated region to the tracker, ptr = %p, size "
+    // check if the allocation was already added to the tracker
+    // (in case of using ProxyLib)
+    tracker_value_t *value =
+        (tracker_value_t *)critnib_get(p->hTracker->map, *(uintptr_t *)ptr);
+    if (value) {
+        assert(value->pool != p->pool);
+
+        LOG_DEBUG("ptr already exists in the tracker (added by Proxy Lib) - "
```

Comment:
Could you please describe the case when memory already present in the tracker?

Reply:
Here or in the code? This is the situation described in https://github.com/oneapi-src/unified-memory-framework/issues/365. Shortly:
* at the beginning of the program the Proxy Lib got a small allocation request (from GTest) but it allocates multiple pages using Jemalloc + OS provider
* next an allocation request is made by the user (a Proxy Pool is used). Proxy Pool calls malloc() which calls Proxy Lib. Jemalloc handles this by calling arenaSplit() which splits large initial allocation into a few smaller ones (no new call to the OS provider). Note that all of them are added to the Tracker
* next the requested allocation is returned by malloc() and here there is NO distinguish between a true new allocation (returned from OS provider) which is not added to the tracker yet, or the allocation created using arenaSplit() which is already in the Tracker
* next the Tracking Provider used in the Proxy Poll adds the allocation to the tracker for the second time - and we have a problem

My proposed solution is to change the "pool" in allocations already tracked to the pool that requested the allocation. I don't have a better idea - @vinser52 @pbalcer @igchor please suggest something better if you can :)

Reply:
Sorry, but did not get what happened.

In the #365 mentioned that `two memory providers together with two tracking providers are stacked, one of them is using the other one`. I did not get how it is possible.

Reply:
@vinser52 the example is Proxy Lib with Jemalloc Pool + OS provider run with Proxy Pool with malloc-based provider, so all malloc calls are handled by Proxy Lib